chrome/content/zotero/itemTree.jsx

```diff
@@ -3002,23 +3002,59 @@ var ItemTree = class ItemTree extends LibraryTree {
 			case 'dateModified':
 			case 'accessDate':
 			case 'date':
-				if (key == 'date' && !this.collectionTreeRow.isFeedsOrFeed()) {
-					break;
-				}
 				if (val) {
-					let date = Zotero.Date.sqlToDate(val, true);
-					if (date) {
-						// If no time, interpret as local, not UTC
-						if (Zotero.Date.isSQLDate(val)) {
-							date = Zotero.Date.sqlToDate(val);
-							val = date.toLocaleDateString();
+					if (key != 'date' || this.collectionTreeRow.isFeedsOrFeed()) {
+						let date = Zotero.Date.sqlToDate(val, true);
+						if (date) {
+							// If no time, interpret as local, not UTC
+							if (Zotero.Date.isSQLDate(val)) {
+								date = Zotero.Date.sqlToDate(val);
+								val = date.toLocaleDateString();
+							}
+							else {
+								val = date.toLocaleString();
+							}
 						}
 						else {
-							val = date.toLocaleString();
+							val = '';
 						}
 					}
 					else {
-						val = '';
+						// key == 'date' and we aren't in a feed
+						// Try to parse the date
+						let { year, month, day } = Zotero.Date.strToDate(val);
+						// See strToMultipart() - discard year if it contains a suffix
+						if (!/^\d{1,4}$/.test(year)) {
+							year = undefined;
+						}
+						year = parseInt(year);
+						if (isNaN(year)) {
+							year = undefined;
+						}
+						month = parseInt(month);
+						if (isNaN(month)) {
+							month = undefined;
+						}
+						day = parseInt(day);
+						if (isNaN(day)) {
+							day = undefined;
+						}
+						if (year !== undefined || month !== undefined || day !== undefined) {
```

Comment:
We want to use the raw value if there's no year. Currently if there's, say, a time in the Date field (`09:32:53`), it will show just a number (`9`), but when sorting by Date that row will be among rows without date fields. (This would also be taken care of by failing parsing when there's a `part` value, though — in this example, part would be `32:53`.)